woocommerce/woocommerce-gutenberg-products-block
# Product Stock indicator: Add support for background color, margin, padding & various typography controls

This Pull Request enhances the styling capabilities of the Product Stock Indicator block by introducing additional support for background color, typography & spacing. This improvement provides users with more flexibility when customizing the appearance of the Product Stock Indicator block.

Here is what Product Stock Indicator block supports now:

**Color:**
- Text color
- Background color

**Typography:**
- Font family 
- Font size
- Font Appearance
- Line height
- Letter spacing 
- Text decoration 
- Letter Case

**Spacing:**
- Margin
- Padding


Part of #7954 & #8695 

### Testing

#### Test using blocks sidebar
- Create a new page or post
- Add the "Products" block to your post.
- Add Product Stock Indicator block as an inner block if not included already
- Select the Product Stock Indicator block and open the block settings in the right sidebar.
- Customize the block using the new styling options available:
   - Change the text color, background color, font family, font size, font appearance, line height, letter spacing, text decoration, letter case, margin, and padding under the appropriate settings sections.
- Verify that the styling changes are immediately visible and correctly applied within the editor.
- Save the post, then view it on the front end to ensure the styling changes are visible and applied correctly.

#### Test using Global Styles
- Open the Global Styles sidebar by clicking the "Styles" button at the top right corner of the editor.

https://user-images.githubusercontent.com/16707866/228781997-24a9a29f-4be7-46d4-b1b3-41520692d86d.mov

- In the Global Styles sidebar, locate the "Product Stock Indicator" block.
- Customize the block using the new styling options available under the "Product Stock Indicator" block in the Global Styles sidebar:
   - Change the text color, background color, font family, font size, font appearance, line height, letter spacing, text decoration, letter case, margin, and padding under the appropriate settings sections.
- Verify that these applied correctly within editor & on frontend:
   - Create a new post
   - Add Products block
   - Add Product Stock Indicator block as an inner block if not included already
   - Check the Product Stock Indicator block in both the editor and on the front-end to ensure that the changes made through the Global styles UI are accurately applied:
       - In the editor, inspect Product Stock Indicator block & ensure it matches the values set in the Global styles UI.
       - Save the post & publish it. On frontend, inspect Product Stock Indicator block & ensure it matches the values set in the Global styles UI.

> Also, verify the changes in All Products block, which also uses Product Stock Indicator block.



### WooCommerce Visibility




### Changelog

> Product Stock indicator: Added support for background color, margin, padding & various typography controls
